### PR TITLE
Set mutated node dict in spec

### PR DESCRIPTION
**Pull request by rabi**

### CODE REVIEW

**Review by fao89 — APPROVED** [review]

**Review by gibizer — APPROVED** [review]
Reproduced the problem without the fix then applied the this fix and redeployed the DataPlane CR. With the fix the ssh secret gets defaulted properly.

**Comment by fao89** [comment]
CI is having issues with webhook on kuttl
```
case.go:364: failed in step 0-dataplane-create
case.go:366: Internal error occurred: failed calling webhook "mopenstackdataplane.kb.io": failed to call webhook: Post "[https://dataplane-operator-controller-manager-service.openstack-operators.svc:443/mutate-dataplane-openstack-org-v1beta1-openstackdataplane?timeout=10s](https://dataplane-operator-controller-manager-service.openstack-operators.svc/mutate-dataplane-openstack-org-v1beta1-openstackdataplane?timeout=10s)": no endpoints available for service "dataplane-operator-controller-manager-service"
```    
https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openstack-k8s-operators_dataplane-operator/279/pull-ci-openstack-k8s-operators-dataplane-operator-main-dataplane-operator-build-deploy-kuttl/1678365753696325632#1:build-log.txt%3A1032

File: api/v1beta1/openstackdataplane_webhook.go [hunk 1]
```diff
@@ -66,6 +66,7 @@ func (spec *OpenStackDataPlaneSpec) Default() {
 			roleSSHSecret := spec.Roles[node.Role].NodeTemplate.AnsibleSSHPrivateKeySecret
 			node.Node.AnsibleSSHPrivateKeySecret = roleSSHSecret
 		}
+		spec.Nodes[nodeName] = *node.DeepCopy()
```
Comment (bshephar) [inline api/v1beta1/openstackdataplane_webhook.go:69]: Can we have the discussion around architecture first?

Under the current architecture, this makes sense. Particularly removing the NetworkConfig j2 file definitions. But do we want to change this setup around nodes in general is my broader question.

**Review by bshephar — COMMENTED** [review]
Can we have the discussion around architecture first?

**Comment by rabi** [comment]
> But do we want to change this setup around nodes in general is my broader question.

Not sure I understand. What do you mean "setup around nodes"? HostName is an optional field from the beginning and it was initialized to node name in reconciliation and now we do in the webhook.  Btw, this patch just fixes an issue with webhooks.

**Comment by rabi** [comment]
Strange, looks like controller-manager does not do anything for 10 mins and the webhooks are not initialized.

https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openstack-k8s-operators_dataplane-operator/279/pull-ci-openstack-k8s-operators-dataplane-operator-main-dataplane-operator-build-deploy-kuttl/1678718358272348160/artifacts/dataplane-operator-build-deploy-kuttl/openstack-k8s-operators-gather/artifacts/controller-manager/dataplane-operator-controller-manager-594df448bc-w7xpk-containers.log

**Comment by rabi** [comment]
Tested locally with the dataplane index image built with that PR and it seems all good. http://pastebin.test.redhat.com/1104634

**Comment by fao89** [comment]
I believe the issue is not with this PR, I'm ok to merge it!

**Pull request merged**